### PR TITLE
Added JS / TS section

### PR DESCRIPTION
**Pull request by vid553**
I've written the section that includes:
- general guidelines in regards to usage of package manager and templates
- link to coding standard and links to additional documentation
- coding style, where I've also added configuration files for all types of projects
- docstring info and example

### CODE REVIEW

File: docs/developer_guidelines.md [hunk 1]
```diff
@@ -363,7 +369,85 @@ comprehensive document refers to the [rustdoc book].
 
 ### JavaScript/TypeScript 🇯🇸 / 🇹🇸
 
-🚧 **In progress** 🚧
+#### General guidelines
+- Use `yarn` instead of `npm` package manager:
+  - `yarn add <package_name>` instead of `npm install <package_name>`,
+  - `yarn add --dev <package_name>` instead of `npm install --save-dev <package_name>`,
+  - `yarn create` instead of `npx`.
+- Use templates for React or RN when creating a new project.
```
Comment (MarkoSagadin) [inline docs/developer_guidelines.md:377]: What is RN?

**Review by MarkoSagadin — CHANGES_REQUESTED** [review]
Thank you for adding this section :)

File: docs/developer_guidelines.md (hunk 1 above)
Comment (vid553) [reply docs/developer_guidelines.md:377]: React Native

**Comment by vid553** [comment]
When you are done editing the markdown document please running through Grammarly - was done

I would like to see that you clean up the commits before merging. There are two options, you rebase/squash locally and force push or select Squash and Merge option when merging PR and add a commit message that is similiar to the one in PR (I think this would be less work, plus it makes this addition atomic.) - squash when merging

Finally, I would suggest that at some later point a irnas-react-template project is created and all config files are moved there.
This was already discussed with Musti and will be done at some point in the future.

**Pull request merged**